corehq/apps/saved_reports/models.py

```diff
@@ -691,84 +695,141 @@ def _get_or_create_uuid(self):
-                    details={
-                        'subject': title,
-                        'recipients': str(emails),
-                        'error': er,
-                    }
-                )
-                if excel_files:
-                    message = _("Unable to generate email report. Excel files are attached.")
-                    send_html_email_async(title, email, message,
-                                          email_from=settings.DEFAULT_FROM_EMAIL,
-                                          file_attachments=excel_files,
-                                          smtp_exception_skip_list=LARGE_FILE_SIZE_ERROR_CODES)
-
-                elif getattr(er, 'smtp_code', None) in LARGE_FILE_SIZE_ERROR_CODES or type(er) == ESError:
-                    # If the email doesn't work because it is too large to fit in the HTML body,
-                    # send it as an excel attachment, by creating a mock request with the right data.
-
-                    for report_config in self.configs:
-                        mock_request = HttpRequest()
-                        mock_request.couch_user = self.owner
-                        mock_request.user = self.owner.get_django_user()
-                        mock_request.domain = self.domain
-                        mock_request.couch_user.current_domain = self.domain
-                        mock_request.couch_user.language = self.language
-                        mock_request.method = 'GET'
-                        mock_request.bypass_two_factor = True
-
-                        mock_query_string_parts = [report_config.query_string, 'filterSet=true']
-                        mock_request.GET = QueryDict('&'.join(mock_query_string_parts))
-                        request_data = vars(mock_request)
-                        request_data['couch_user'] = mock_request.couch_user.userID
-                        if report_config.report_slug != ApplicationStatusReport.slug:
-                            # ApplicationStatusReport doesn't have date filter
-                            date_range = report_config.get_date_range()
-                            start_date = datetime.strptime(date_range['startdate'], '%Y-%m-%d')
-                            end_date = datetime.strptime(date_range['enddate'], '%Y-%m-%d')
-                            datespan = DateSpan(start_date, end_date)
-                            request_data['datespan'] = datespan
-
-                        full_request = {'request': request_data,
-                                        'domain': request_data['domain'],
-                                        'context': {},
-                                        'request_params': json_request(request_data['GET'])}
-
-                        export_all_rows_task(report_config.report, full_request, emails, title)
+                if isinstance(er, SMTPSenderRefused) and (er.smtp_code in LARGE_FILE_SIZE_ERROR_CODES):
```

Comment:
unnecessary parens
```suggestion
                if isinstance(er, SMTPSenderRefused) and er.smtp_code in LARGE_FILE_SIZE_ERROR_CODES:
```

Reply:
I think the explicit parens make this a more readable expression, but I think we've disagreed on this before. I feel like no parens tends to read like a run-on sentence (especially with several clauses) and forces me to look at the entire statement to determine what precedence is being used. Leaving this as is for now.